### PR TITLE
Fix crash with using 'replce property with method' with interfaces in VB

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://devdiv.visualstudio.com/DevDiv/NET%20Developer%20Experience%20IDE/_workitems?_a=edit&id=410568&triage=true
Fixes https://devdiv.visualstudio.com/DevDiv/NET%20Developer%20Experience%20IDE/_workitems?_a=edit&id=440371&triage=true

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide @Pilchie

**Comment by CyrusNajmabadi** [comment]
Checking out 'replace method with property' now to see if we have similar issues.

**Comment by CyrusNajmabadi** [comment]
retest this please

**Comment by CyrusNajmabadi** [comment]
retest windows_debug_vs-integration_prtest please

**Comment by CyrusNajmabadi** [comment]
retest windows_release_vs-integration_prtest please

File: src/Features/VisualBasic/Portable/ReplacePropertyWithMethods/VisualBasicReplacePropertyWithMethods.vb [hunk 1]
```diff
@@ -150,11 +153,43 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.CodeRefactorings.ReplaceMethodWithP
...
             Return methodDeclaration
         End Function
 
+        Private Function UpdateExplicitInterfaceImplementations(
+                [property] As IPropertySymbol,
+                method As IMethodSymbol,
+                desiredName As String) As IMethodSymbol
+
+            Dim updatedExplicitImpl = UpdateExplicitInterfaceImplementation(
+                [property], method.ExplicitInterfaceImplementations.FirstOrDefault(), desiredName)
```
Comment (dpoeschl) [inline src/Features/VisualBasic/Portable/ReplacePropertyWithMethods/VisualBasicReplacePropertyWithMethods.vb]: To me, `FirstOrDefault` is always either a bug or deserves a comment.

File: src/Features/VisualBasic/Portable/ReplacePropertyWithMethods/VisualBasicReplacePropertyWithMethods.vb [hunk 2]
```diff
@@ -150,11 +153,43 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.CodeRefactorings.ReplaceMethodWithP
...
+
+            Return If(updatedExplicitImpl Is method.ExplicitInterfaceImplementations.FirstOrDefault(),
+                      method,
+                      CodeGenerationSymbolFactory.CreateMethodSymbol(
+                        method, explicitInterfaceSymbol:=updatedExplicitImpl))
+        End Function
+
+        Private Function UpdateExplicitInterfaceImplementation(
+                [property] As IPropertySymbol,
+                methodSymbol As IMethodSymbol,
```
Comment (dpoeschl) [inline src/Features/VisualBasic/Portable/ReplacePropertyWithMethods/VisualBasicReplacePropertyWithMethods.vb]: `explicitlyImplementedMethodSymbol`?

File: src/Features/VisualBasic/Portable/ReplacePropertyWithMethods/VisualBasicReplacePropertyWithMethods.vb (hunk 1 above)
Comment (dpoeschl) [inline src/Features/VisualBasic/Portable/ReplacePropertyWithMethods/VisualBasicReplacePropertyWithMethods.vb]: (we talked about why it's like this, a limitation of the symbol generation apis, so I signed off on the assumption that you'll add a comment)

**Comment by CyrusNajmabadi** [comment]
Retest this please

**Comment by CyrusNajmabadi** [comment]
Retesting for nuget 0x00 problem.

**Comment by CyrusNajmabadi** [comment]
Tagging @MattGertz   Fixes two high hitting crashes in the IDE.


**Customer scenario**

Customer uses the "replace property with method" code fix on a VB interface property and gets a crash.

**Bugs this fixes:**

Fixes https://devdiv.visualstudio.com/DevDiv/NET%20Developer%20Experience%20IDE/_workitems?_a=edit&id=410568&triage=true
Fixes https://devdiv.visualstudio.com/DevDiv/NET%20Developer%20Experience%20IDE/_workitems?_a=edit&id=440371&triage=true

**Workarounds, if any**

none.

**Risk**

Low.

**Performance impact**

Low.

**Is this a regression from a previous update?**

No.

**Root cause analysis:**

Non exhaustive testing of all language permutations against feature. 

**How was the bug found?**

Customer watsons.

**Comment by dpoeschl** [comment]
@CyrusNajmabadi 

```
Tagging @MattGertz Fixes two high hitting crashes in the IDE.

Customer uses the "replace property with method" code fix on a VB interface property and gets a crash.
```

It's not *crashing*, right? Just giving the info bar?

**Comment by CyrusNajmabadi** [comment]
Yes.  Sorry, NFW crashes.

**Comment by sophistson** [comment]
How can I get this fix ? At the moment I have no rights to download it: 
" (Microsoft account) is not authorized to access this page"

**Comment by Pilchie** [comment]
Those are just internal tracking bugs, they aren't downloads.  This fix will appear as part of Visual Studio 2017 Preview version 15.3, but isn't there in the current public builds.

**Comment by sophistson** [comment]
ok, thx

**Pull request merged**